best-practices/puppet_prs.md

```diff
@@ -0,0 +1,28 @@
+# Puppet PRs
+
+Occasionally, a developer may need a system level configuration change made to a machine. Perhaps the most common example is when a dev needs ssh access to a machine to be able to kick off a capistrano task. In this case, we encourage devs to make a pull request to the Puppet repository where such configuration is mannaged.
+
+To issue a PR to the DLSS puppet repository, first make sure you have a local clone of the repo. Inside, check out a branch off of `production` on which to do your work. By convention the name of the branch should include your sunetid, so that operations knows who is issuing the PR:
+
+```
+git checkout -b sunetidFeature
+git branch --set-upstream-to=origin/production  
```

Comment:
i'm not totally sure, but wouldn't this result in the feature branch pushing directly to the production branch by default?  this might just be a hole in my git knowledge.

fwiw, what i'd usually do is set the upstream when i push, something like:
```
git push -u origin sunetidFeature
```

which will set your local `sunetidFeature` branch to track to the remote `sunetIdFeature` branch (and to create that branch on origin when it's first pushed, allowing the `-u` to be omitted on subsequent pushes.